src/layers/core/grid-layer/grid-layer.js

```diff
@@ -84,21 +123,48 @@ export default class GridLayer extends CompositeLayer {
     });
   }
 
+  getUpdateTriggers() {
+    return {
+      getColor: {
+        colorRange: this.props.colorRange,
+        colorDomain: this.props.colorDomain,
+        getColorValue: this.props.getColorValue,
+        lowerPercentile: this.props.lowerPercentile,
+        upperPercentile: this.props.upperPercentile
+      },
+      getElevation: {
+        elevationRange: this.props.elevationRange,
+        elevationDomain: this.props.elevationDomain
+      }
+    };
+  }
+
   _onGetSublayerColor(cell) {
     const {colorRange} = this.props;
-    const colorDomain = this.props.colorDomain || this.state.countRange;
+    const {valueDomain, sortedBins} = this.state;
+    const value = sortedBins.binMap[cell.index] && sortedBins.binMap[cell.index].value;
+
+    const colorDomain = this.props.colorDomain || valueDomain;
+    const color = quantizeScale(colorDomain, colorRange, value);
+
+    // if cell value is outside domain, set alpha to 0
+    const alpha = value >= valueDomain[0] && value <= valueDomain[1] ?
+      (Number.isFinite(color[3]) ? color[3] : 255) : 0;
 
-    return quantizeScale(colorDomain, colorRange, cell.count);
+    return color.concat([alpha]);
```

Comment:
Not sure how frequently this is called, but cloning arrays (Array.concat) seems like a recipe for bad performance. Can `quantizeScale` just take an `alpha` argument?